arangod/StorageEngine/StorageEngine.cpp

```diff
@@ -61,9 +61,18 @@ Result StorageEngine::writeCreateDatabaseMarker(TRI_voc_tick_t id,
 }
 
 Result StorageEngine::prepareDropDatabase(TRI_vocbase_t& vocbase) { return {}; }
+
 bool StorageEngine::inRecovery() {
   return recoveryState() < RecoveryState::DONE;
 }
+
+void StorageEngine::scheduleFullIndexRefill(std::string const& database,
+                                            std::string const& collection,
+                                            IndexId iid) {
+  // should not be called on the base engine
+  TRI_ASSERT(false);
```

Comment:
Should we make it pure virtual?

Reply:
I don't think we should make it pure virtual, because that may trigger runtime termination of the process. `TRI_ASSERT(false);` will only trigger in maintainer mode, but in production it will not cause any unintended trouble.